pkg/apis/azureprovider/v1beta1/types.go

```diff
@@ -401,11 +401,17 @@ type Image struct {
 type VMIdentity string
 
 type OSDisk struct {
-	OSType      string      `json:"osType"`
-	ManagedDisk ManagedDisk `json:"managedDisk"`
-	DiskSizeGB  int32       `json:"diskSizeGB"`
+	OSType      string                `json:"osType"`
+	ManagedDisk ManagedDiskParameters `json:"managedDisk"`
+	DiskSizeGB  int32                 `json:"diskSizeGB"`
 }
 
-type ManagedDisk struct {
-	StorageAccountType string `json:"storageAccountType"`
+type ManagedDiskParameters struct {
+	StorageAccountType string                       `json:"storageAccountType"`
+	DiskEncryptionSet  *DiskEncryptionSetParameters `json:"diskEncryptionSet,omitempty"`
+	ID                 string                       `json:"id,omitempty"`
```

Comment:
Is this ID used at all? I can't see where.

Reply:
I believe it is used here:
https://github.com/openshift/cluster-api-provider-azure/pull/158/files#diff-79058a3142fd624b3690ff1dc72c0d2cR193

Reply:
@enxebre, sorry I thought this referenced the `DiskEncryptionSet.ID`.  I included this `ID` because it is found inside of the `ManagedDiskParameters` here https://github.com/Azure/azure-sdk-for-go/blob/master/services/compute/mgmt/2019-12-01/compute/models.go#L6507.  In case it is ever parsed or assigned the struct here would have a reference to it.

Reply:
Let's not leave in unused fields so we don't have to worry about a user putting something wrong in there.

Reply:
Removed